src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,410 @@
+    quoteToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+    amount: '1234567890',
+    kind: OrderKind.BUY,
+  } as PriceQuoteParams)
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/rinkeby/api/v1/markets/0x6810e776880c02933d47db1b9fc05908e5386b96-0x6810e776880c02933d47db1b9fc05908e5386b96/buy/1234567890',
+    FETCH_RESPONSE_PARAMETERS
+  )
+  expect(price?.amount).toEqual(PRICE_QUOTE_RESPONSE.amount)
+  expect(price?.token).toEqual(PRICE_QUOTE_RESPONSE.token)
+})
+
+test('Invalid: Get Price Quote (Legacy) with unexisting token', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  try {
+    await cowSdk.cowApi.getPriceQuoteLegacy({
+      baseToken: 'unexistingToken',
+      quoteToken: 'unexistingToken',
+      amount: '1234567890',
+      kind: OrderKind.BUY,
+    } as PriceQuoteParams)
+  } catch (e) {
+    const error = e as QuoteError
+    expect(error.message).toEqual('Token pair selected has insufficient liquidity')
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith(
+      'https://api.cow.fi/rinkeby/api/v1/markets/unexistingToken-unexistingToken/buy/1234567890',
+      FETCH_RESPONSE_PARAMETERS
+    )
+  }
+})
+
+test('Valid: Get Profile Data', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(PROFILE_DATA_RESPONSE), { status: HTTP_STATUS_OK })
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  await cowSdk1.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/affiliate/api/v1/profile/0x6810e776880c02933d47db1b9fc05908e5386b96',
+    FETCH_RESPONSE_PARAMETERS
+  )
+})
+
+test('Invalid: Get Profile Data from unexisting address', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  try {
+    await cowSdk1.cowApi.getProfileData('unexistingAddress')
+  } catch (e) {
+    const error = e as CowError
+    expect(error.message).toEqual("You've passed an invalid URL")
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith(
+      'https://api.cow.fi/affiliate/api/v1/profile/unexistingAddress',
+      FETCH_RESPONSE_PARAMETERS
+    )
+  }
+})
+
+test('Invalid: Get Profile Data from not supported network', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(PROFILE_DATA_RESPONSE), { status: HTTP_STATUS_OK })
+  const profileData = await cowSdk.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
+  expect(profileData).toBeNull()
```

Comment:
Sorry for the dumb question, but I still don't get how you get a null response here.
As far as I see, the mock will return a valid response and it doesn't require it to match the query string. Or does it?

Reply:
yes, this test is a bit misleading, the profile data is only available for mainnet, and the sdk was instantiated for rinkeby, there's currently an [early return](https://github.com/cowprotocol/cow-sdk/blob/main/src/api/cow/index.ts#L113-L115) that handles that. But yes, the test description could be improved so it's obvious.

Reply:
Ahhh right, now it makes sense.
A few comments would definitely help to make this clearer in the test